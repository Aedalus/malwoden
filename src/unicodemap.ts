import { CharCode } from "./char-code";

export const unicodeMap: { [index: number]: number } = {
  // 1 - 15.
  [CharCode.whiteSmilingFace]: 1,
  [CharCode.blackSmilingFace]: 2,
  [CharCode.blackHeartSuit]: 3,
  [CharCode.blackDiamondSuit]: 4,
  [CharCode.blackClubSuit]: 5,
  [CharCode.blackSpadeSuit]: 6,
  [CharCode.bullet]: 7,
  [CharCode.inverseBullet]: 8,
  [CharCode.whiteCircle]: 9,
  [CharCode.inverseWhiteCircle]: 10,
  [CharCode.maleSign]: 11,
  [CharCode.femaleSign]: 12,
  [CharCode.eighthNote]: 13,
  [CharCode.beamedEighthNotes]: 14,
  [CharCode.whiteSunWithRays]: 15,

  // 16 - 31.
  [CharCode.blackRightPointingPointer]: 16,
  [CharCode.blackLeftPointingPointer]: 17,
  [CharCode.upDownArrow]: 18,
  [CharCode.doubleExclamationMark]: 19,
  [CharCode.pilcrow]: 20,
  [CharCode.sectionSign]: 21,
  [CharCode.blackRectangle]: 22,
  [CharCode.upDownArrowWithBase]: 23,
  [CharCode.upwardsArrow]: 24,
  [CharCode.downwardsArrow]: 25,
  [CharCode.rightwardsArrow]: 26,
  [CharCode.leftwardsArrow]: 27,
  [CharCode.rightAngle]: 28,
  [CharCode.leftRightArrow]: 29,
  [CharCode.blackUpPointingTriangle]: 30,
  [CharCode.blackDownPointingTriangle]: 31,

  // 127.
  [CharCode.house]: 127,

  // 128 - 143.
  [CharCode.latinCapitalLetterCWithCedilla]: 128,
  [CharCode.latinSmallLetterUWithDiaeresis]: 129,
  [CharCode.latinSmallLetterEWithAcute]: 130,
  [CharCode.latinSmallLetterAWithCircumflex]: 131,
  [CharCode.latinSmallLetterAWithDiaeresis]: 132,
  [CharCode.latinSmallLetterAWithGrave]: 133,
  [CharCode.latinSmallLetterAWithRingAbove]: 134,
  [CharCode.latinSmallLetterCWithCedilla]: 135,
  [CharCode.latinSmallLetterEWithCircumflex]: 136,
  [CharCode.latinSmallLetterEWithDiaeresis]: 137,
  [CharCode.latinSmallLetterEWithGrave]: 138,
  [CharCode.latinSmallLetterIWithDiaeresis]: 139,
  [CharCode.latinSmallLetterIWithCircumflex]: 140,
  [CharCode.latinSmallLetterIWithGrave]: 141,
  [CharCode.latinCapitalLetterAWithDiaeresis]: 142,
  [CharCode.latinCapitalLetterAWithRingAbove]: 143,

  // 144 - 159.
  [CharCode.latinCapitalLetterEWithAcute]: 144,
  [CharCode.latinSmallLetterAe]: 145,
  [CharCode.latinCapitalLetterAe]: 146,
  [CharCode.latinSmallLetterOWithCircumflex]: 147,
  [CharCode.latinSmallLetterOWithDiaeresis]: 148,
  [CharCode.latinSmallLetterOWithGrave]: 149,
  [CharCode.latinSmallLetterUWithCircumflex]: 150,
  [CharCode.latinSmallLetterUWithGrave]: 151,
  [CharCode.latinSmallLetterYWithDiaeresis]: 152,
  [CharCode.latinCapitalLetterOWithDiaeresis]: 153,
  [CharCode.latinCapitalLetterUWithDiaeresis]: 154,
  [CharCode.centSign]: 155,
  [CharCode.poundSign]: 156,
  [CharCode.yenSign]: 157,
  [CharCode.pesetaSign]: 158,
  [CharCode.latinSmallLetterFWithHook]: 159,

  // 160 - 175.
  [CharCode.latinSmallLetterAWithAcute]: 160,
  [CharCode.latinSmallLetterIWithAcute]: 161,
  [CharCode.latinSmallLetterOWithAcute]: 162,
  [CharCode.latinSmallLetterUWithAcute]: 163,
  [CharCode.latinSmallLetterNWithTilde]: 164,
  [CharCode.latinCapitalLetterNWithTilde]: 165,
  [CharCode.feminineOrdinalIndicator]: 166,
  [CharCode.masculineOrdinalIndicator]: 167,
  [CharCode.invertedQuestionMark]: 168,
  [CharCode.reversedNotSign]: 169,
  [CharCode.notSign]: 170,
  [CharCode.vulgarFractionOneHalf]: 171,
  [CharCode.vulgarFractionOneQuarter]: 172,
  [CharCode.invertedExclamationMark]: 173,
  [CharCode.leftPointingDoubleAngleQuotationMark]: 174,
  [CharCode.rightPointingDoubleAngleQuotationMark]: 175,

  // 176 - 191.
  [CharCode.lightShade]: 176,
  [CharCode.mediumShade]: 177,
  [CharCode.darkShade]: 178,
  [CharCode.boxDrawingsLightVertical]: 179,
  [CharCode.boxDrawingsLightVerticalAndLeft]: 180,
  [CharCode.boxDrawingsVerticalSingleAndLeftDouble]: 181,
  [CharCode.boxDrawingsVerticalDoubleAndLeftSingle]: 182,
  [CharCode.boxDrawingsDownDoubleAndLeftSingle]: 183,
  [CharCode.boxDrawingsDownSingleAndLeftDouble]: 184,
  [CharCode.boxDrawingsDoubleVerticalAndLeft]: 185,
  [CharCode.boxDrawingsDoubleVertical]: 186,
  [CharCode.boxDrawingsDoubleDownAndLeft]: 187,
  [CharCode.boxDrawingsDoubleUpAndLeft]: 188,
  [CharCode.boxDrawingsUpDoubleAndLeftSingle]: 189,
  [CharCode.boxDrawingsUpSingleAndLeftDouble]: 190,
  [CharCode.boxDrawingsLightDownAndLeft]: 191,

  // 192 - 207.
  [CharCode.boxDrawingsLightUpAndRight]: 192,
  [CharCode.boxDrawingsLightUpAndHorizontal]: 193,
  [CharCode.boxDrawingsLightDownAndHorizontal]: 194,
  [CharCode.boxDrawingsLightVerticalAndRight]: 195,
  [CharCode.boxDrawingsLightHorizontal]: 196,
  [CharCode.boxDrawingsLightVerticalAndHorizontal]: 197,
  [CharCode.boxDrawingsVerticalSingleAndRightDouble]: 198,
  [CharCode.boxDrawingsVerticalDoubleAndRightSingle]: 199,
  [CharCode.boxDrawingsDoubleUpAndRight]: 200,
  [CharCode.boxDrawingsDoubleDownAndRight]: 201,
  [CharCode.boxDrawingsDoubleUpAndHorizontal]: 202,
  [CharCode.boxDrawingsDoubleDownAndHorizontal]: 203,
  [CharCode.boxDrawingsDoubleVerticalAndRight]: 204,
  [CharCode.boxDrawingsDoubleHorizontal]: 205,
  [CharCode.boxDrawingsDoubleVerticalAndHorizontal]: 206,
  [CharCode.boxDrawingsUpSingleAndHorizontalDouble]: 207,

  // 208 - 223.
  [CharCode.boxDrawingsUpDoubleAndHorizontalSingle]: 208,
  [CharCode.boxDrawingsDownSingleAndHorizontalDouble]: 209,
  [CharCode.boxDrawingsDownDoubleAndHorizontalSingle]: 210,
  [CharCode.boxDrawingsUpDoubleAndRightSingle]: 211,
  [CharCode.boxDrawingsUpSingleAndRightDouble]: 212,
  [CharCode.boxDrawingsDownSingleAndRightDouble]: 213,
  [CharCode.boxDrawingsDownDoubleAndRightSingle]: 214,
  [CharCode.boxDrawingsVerticalDoubleAndHorizontalSingle]: 215,
  [CharCode.boxDrawingsVerticalSingleAndHorizontalDouble]: 216,
  [CharCode.boxDrawingsLightUpAndLeft]: 217,
  [CharCode.boxDrawingsLightDownAndRight]: 218,
  [CharCode.fullBlock]: 219,
  [CharCode.lowerHalfBlock]: 220,
  [CharCode.leftHalfBlock]: 221,
  [CharCode.rightHalfBlock]: 222,
  [CharCode.upperHalfBlock]: 223,

  // 224 - 239.
  [CharCode.greekSmallLetterAlpha]: 224,
  [CharCode.latinSmallLetterSharpS]: 225,
  [CharCode.greekCapitalLetterGamma]: 226,
  [CharCode.greekSmallLetterPi]: 227,
  [CharCode.greekCapitalLetterSigma]: 228,
  [CharCode.greekSmallLetterSigma]: 229,
  [CharCode.microSign]: 230,
  [CharCode.greekSmallLetterTau]: 231,
  [CharCode.greekCapitalLetterPhi]: 232,
  [CharCode.greekCapitalLetterTheta]: 233,
  [CharCode.greekCapitalLetterOmega]: 234,
  [CharCode.greekSmallLetterDelta]: 235,
  [CharCode.infinity]: 236,
  [CharCode.greekSmallLetterPhi]: 237,
  [CharCode.greekSmallLetterEpsilon]: 238,
  [CharCode.intersection]: 239,

  // 240 - 255.
  [CharCode.identicalTo]: 240,
  [CharCode.plusMinusSign]: 241,
  [CharCode.greaterThanOrEqualTo]: 242,
  [CharCode.lessThanOrEqualTo]: 243,
  [CharCode.topHalfIntegral]: 244,
  [CharCode.bottomHalfIntegral]: 245,
  [CharCode.divisionSign]: 246,
  [CharCode.almostEqualTo]: 247,
  [CharCode.degreeSign]: 248,
  [CharCode.bulletOperator]: 249,
  [CharCode.middleDot]: 250,
  [CharCode.squareRoot]: 251,
  [CharCode.superscriptLatinSmallLetterN]: 252,
  [CharCode.superscriptTwo]: 253,
  [CharCode.blackSquare]: 254
};
