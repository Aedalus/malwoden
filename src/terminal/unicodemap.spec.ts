import { unicodeMap } from "./unicodemap";

describe("unicodemap", () => {
  it("Should have unicode values", () => {
    expect(unicodeMap).toMatchSnapshot();
  });
});
