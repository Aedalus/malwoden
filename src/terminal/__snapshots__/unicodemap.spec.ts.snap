// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unicodemap Should have unicode values 1`] = `
Object {
  "161": 173,
  "162": 155,
  "163": 156,
  "165": 157,
  "167": 21,
  "170": 166,
  "171": 174,
  "172": 170,
  "176": 248,
  "177": 241,
  "178": 253,
  "181": 230,
  "182": 20,
  "183": 250,
  "186": 167,
  "187": 175,
  "188": 172,
  "189": 171,
  "191": 168,
  "196": 142,
  "197": 143,
  "198": 146,
  "199": 128,
  "201": 144,
  "209": 165,
  "214": 153,
  "220": 154,
  "223": 225,
  "224": 133,
  "225": 160,
  "226": 131,
  "228": 132,
  "229": 134,
  "230": 145,
  "231": 135,
  "232": 138,
  "233": 130,
  "234": 136,
  "235": 137,
  "236": 141,
  "237": 161,
  "238": 140,
  "239": 139,
  "241": 164,
  "242": 149,
  "243": 162,
  "244": 147,
  "246": 148,
  "247": 246,
  "249": 151,
  "250": 163,
  "251": 150,
  "252": 129,
  "255": 152,
  "402": 159,
  "8226": 7,
  "8252": 19,
  "8319": 252,
  "8359": 158,
  "8592": 27,
  "8593": 24,
  "8594": 26,
  "8595": 25,
  "8596": 29,
  "8597": 18,
  "8616": 23,
  "8729": 249,
  "8730": 251,
  "8734": 236,
  "8735": 28,
  "8745": 239,
  "8776": 247,
  "8801": 240,
  "8804": 243,
  "8805": 242,
  "8962": 127,
  "8976": 169,
  "8992": 244,
  "8993": 245,
  "915": 226,
  "920": 233,
  "931": 228,
  "934": 232,
  "937": 234,
  "945": 224,
  "9472": 196,
  "9474": 179,
  "948": 235,
  "9484": 218,
  "9488": 191,
  "949": 238,
  "9492": 192,
  "9496": 217,
  "9500": 195,
  "9508": 180,
  "9516": 194,
  "9524": 193,
  "9532": 197,
  "9552": 205,
  "9553": 186,
  "9554": 213,
  "9555": 214,
  "9556": 201,
  "9557": 184,
  "9558": 183,
  "9559": 187,
  "9560": 212,
  "9561": 211,
  "9562": 200,
  "9563": 190,
  "9564": 189,
  "9565": 188,
  "9566": 198,
  "9567": 199,
  "9568": 204,
  "9569": 181,
  "9570": 182,
  "9571": 185,
  "9572": 209,
  "9573": 210,
  "9574": 203,
  "9575": 207,
  "9576": 208,
  "9577": 202,
  "9578": 216,
  "9579": 215,
  "9580": 206,
  "960": 227,
  "9600": 223,
  "9604": 220,
  "9608": 219,
  "9612": 221,
  "9616": 222,
  "9617": 176,
  "9618": 177,
  "9619": 178,
  "963": 229,
  "9632": 254,
  "964": 231,
  "9644": 22,
  "9650": 30,
  "9658": 16,
  "966": 237,
  "9660": 31,
  "9668": 17,
  "9675": 9,
  "9688": 8,
  "9689": 10,
  "9786": 1,
  "9787": 2,
  "9788": 15,
  "9792": 12,
  "9794": 11,
  "9824": 6,
  "9827": 5,
  "9829": 3,
  "9830": 4,
  "9834": 13,
  "9835": 14,
}
`;
